### PR TITLE
risc-v/espressif: Add support for RTC subsystem

### PR DESCRIPTION
**Pull request by gustavonihei**
## Summary

This PR intends to enable the NuttX RTC subsystem for the currently supported Espressif RISC-V based chips:
- **ESP32-C3**
- **ESP32-C6**
- **ESP32-H2**

## Impact

New feature, no impact if not explicitly enabled.

## Testing
Successful execution of the `alarm` application with the following defconfigs:
- `esp32c3-generic:rtc`
- `esp32c6-generic:rtc`
- `esp32h2-generic:rtc`

```
nsh> alarm 3
alarm_daemon started
alarm_daemon: Running
Opening /dev/rtc0
Alarm 0 set in 3 seconds
nsh> alarm_daemon: alarm 0 received

```

### CODE REVIEW

**Review by acassis — APPROVED** [review]

**Review by pkarashchenko — COMMENTED** [review]
I've started review, but Alan was faster with merge. I will take a look to other files on Monday

File: arch/risc-v/src/espressif/esp_rtc.c [hunk 1]
```diff
@@ -0,0 +1,878 @@
...
+  struct esp_rtc_lowerhalf_s *priv = (struct esp_rtc_lowerhalf_s *)lower;
+  struct alm_cbinfo_s *cbinfo;
+  uint64_t timeout;
+  irqstate_t flags;
+  int id;
+
+  DEBUGASSERT(priv != NULL);
+  DEBUGASSERT(alarminfo != NULL);
+
+  timeout = mktime((struct tm *)&alarminfo->time);
```
Comment (pkarashchenko) [inline arch/risc-v/src/espressif/esp_rtc.c:433]: ditto

File: arch/risc-v/src/espressif/esp_rtc.c [hunk 2]
```diff
@@ -0,0 +1,878 @@
...
+  irqstate_t flags;
+
+  DEBUGASSERT(lower != NULL && alarminfo != NULL);
+
+  if (alarminfo->reltime > 0)
+    {
+      flags = spin_lock_irqsave(NULL);
+
+      seconds = alarminfo->reltime;
+      gmtime_r(&seconds, (struct tm *)&setalarm.time);
```
Comment (pkarashchenko) [inline arch/risc-v/src/espressif/esp_rtc.c:514]: Maybe we should drop `const` from `alarminfo` instead?

File: arch/risc-v/src/espressif/esp_rtc.c [hunk 3]
```diff
@@ -0,0 +1,878 @@
...
+  /* Get the alarm according to the alarm ID */
+
+  cbinfo = &priv->alarmcb[alarminfo->id];
+
+  ts.tv_sec = (esp_hr_timer_time_us() + g_rtc_save->offset +
+              cbinfo->deadline_us) / USEC_PER_SEC;
+  ts.tv_nsec = ((esp_hr_timer_time_us() + g_rtc_save->offset +
+              cbinfo->deadline_us) % USEC_PER_SEC) * NSEC_PER_USEC;
+
+  localtime_r((const time_t *)&ts.tv_sec,
```
Comment (pkarashchenko) [inline arch/risc-v/src/espressif/esp_rtc.c:628]: type cast is not needed here

File: arch/risc-v/src/espressif/esp_rtc.c [hunk 4]
```diff
@@ -0,0 +1,878 @@
...
+{
+  uint64_t now_us;
+  uint64_t rtc_offset_us;
+  irqstate_t flags;
+
+  DEBUGASSERT(ts != NULL && ts->tv_nsec < NSEC_PER_SEC);
+
+  flags = spin_lock_irqsave(NULL);
+
+  now_us = ((uint64_t) ts->tv_sec) * USEC_PER_SEC +
```
Comment (pkarashchenko) [inline arch/risc-v/src/espressif/esp_rtc.c:773]: remove space from `(uint64_t) ts->tv_sec`

**Pull request merged**